src/main/java/data/schedule/DayList.java

```diff
@@ -0,0 +1,265 @@
+package data.schedule;
+
+import data.plans.Plan;
+import data.plans.PlanList;
+import werkit.Parser;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static data.schedule.InvalidScheduleException.INVALID_DAY;
+
+public class DayList {
+    private static final int FIXED_LENGTH = 30;
+    public static final int NUMBER_OF_SCHEDULE_DAYS = 7;
+
+    private PlanList planList;
+    private String[] scheduleList = new String[NUMBER_OF_SCHEDULE_DAYS];
+    private String[] schedulePrintList = new String[NUMBER_OF_SCHEDULE_DAYS];
+    private Day[] dayList = new Day[NUMBER_OF_SCHEDULE_DAYS]; //store all created day object
+
+    private static Logger logger = Logger.getLogger(Parser.class.getName());
+
+    public DayList(PlanList planList) {
+        this.planList = planList;
+    }
+
+    public String[] getScheduleList() {
+        return scheduleList;
+    }
+
+    public String[] getSchedulePrintList() {
+        return schedulePrintList;
+    }
+
+    public Day[] getDayList() {
+        return this.dayList;
+    }
+
+    /**
+     * Add/Update a plan to a particular day of the week. User specify the plan (plan number) that they
+     * want to add to a particular day (day number). A day object will be created if the day object does not exist,
+     * otherwise the day object will be updated to reflect the latest changes.
+     *
+     * @param userArgument Day number and plan number to be added into the schedule.
+     * @return newDay Day object created with the day number and plan name specified by user.
+     * @throws ArrayIndexOutOfBoundsException If userArgument contains insufficient arguments and parsing fails.
+     * @throws InvalidScheduleException If the schedule parameters specified in userArgument is invalid.
+     */
+    public Day updateDay(String userArgument) throws ArrayIndexOutOfBoundsException, InvalidScheduleException {
```

Comment:
SLAP violation here.
You are doing low level actions (e.g. spliut, parseInt) and high level at once